File: showcase/test-samples.py
# Python sample code
def greet(name):
    message = f"Hello, {name}!"
    print(message)
    return message

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def say_hello(self):
        return f"Hi, I'm {self.name}"

person = Person("Alice", 30)
greet(person.name)

# Keywords: def, class, return, print
# Strings: "Hello, {name}!", "Hi, I'm {self.name}", "Alice"
# Comments: # Python sample code, # Keywords: ...
# Functions: greet, say_hello, __init__
# Variables: message, person, name, age
# Numbers: 30
# Operators: =, ., f"", !
# Classes: Person
# Constants: none

File: showcase/test-samples.go
// Go sample code
package main

import "fmt"

func greet(name string) string {
    message := fmt.Sprintf("Hello, %s!", name)
    fmt.Println(message)
    return message
}

type Person struct {
    Name string
    Age  int
}

func (p Person) SayHello() string {
    return fmt.Sprintf("Hi, I'm %s", p.Name)
}

func main() {
    person := Person{Name: "Alice", Age: 30}
    greet(person.Name)
}

// Keywords: package, import, func, type, return, fmt
// Strings: "Hello, %s!", "Hi, I'm %s", "Alice"
// Comments: // Go sample code, // Keywords: ...
// Functions: greet, SayHello, main
// Variables: message, person, name, Name, Age
// Numbers: 30
// Operators: =, ., :, !, %s
// Classes: Person (struct)
// Constants: none